conans/client/build/cmake_flags.py

```diff
@@ -301,7 +301,7 @@ def get_definitions(self):
                 definitions['CONAN_CXX_FLAGS'] = flag
                 definitions['CONAN_C_FLAGS'] = flag
         else:  # arch_flag is only set for non Visual Studio
-            arch_flag = architecture_flag(compiler=compiler, os=os_, arch=arch)
+            arch_flag = architecture_flag(self._conanfile.settings)
```

Comment:
This means that this PR is not only for Autotools and Meson, but also affects CMake

Reply:
more or less it affects all the build helpers using these `compiler_flags` modules (like `architecture_flag` and so on). the main target was `AutoToolsBuildEnvironment`, but since code is shared (which is good thing), other helpers may also benefit.

Reply:
Sure, not a problem, just a note because I was confused by the PR title.